hca/upload/media_type.py

```diff
@@ -0,0 +1,90 @@
+import magic
+import os
+import re
+
+
+class MediaType:
+
+    RFC7320_TOKEN_CHARSET = "!#$%&'*+-.^_`|~\w"
+    DCP_METADATA_JSON_FILES = ('assay', 'project', 'sample')
+
+    @classmethod
+    def from_string(cls, media_type):
+        split_on_semicolons = media_type.split(';')
+        type_info = split_on_semicolons.pop(0)
+        type_info = re.match("(?P<top_level_type>.+)/(?P<subtype>[^+]+)(?P<suffix>\+.+)?", type_info)
+        parameters = {}
+        for parameter in split_on_semicolons:
+            param_match = re.match('(.+)=(.+)', parameter.strip())
+            parameters[param_match.group(1)] = param_match.group(2).strip('"')
+        return cls(
+            type_info.group('top_level_type'), type_info.group('subtype'),
+            suffix=type_info.group('suffix'), parameters=parameters)
+
+    @classmethod
+    def from_file(cls, file_path, dcp_type=None):
+        media_type_string = cls._media_type_from_magic(file_path)
+        media_type = cls.from_string(media_type_string)
+        if dcp_type:
+            media_type.parameters['dcp-type'] = dcp_type
+        else:
+            media_type.parameters['dcp-type'] = cls._dcp_media_type_param(media_type, os.path.basename(file_path))
+        return media_type
+
+    def __init__(self, top_level_type, subtype, suffix=None, parameters={}):
+        self.top_level_type = top_level_type
+        self.subtype = subtype
+        self.suffix = suffix
+        self.parameters = parameters
+
+    def __str__(self):
+        media_type = self.main_type
+        if self.parameters:
+            for k, v in self.parameters.items():
+                media_type += '; {key}={value}'.format(key=k, value=self._properly_quoted_parameter_value(v))
+        return media_type
```

Comment:
suffix does not figure into here?

Reply:
With `media_type = self.main_type`
`main-type` is an @property that includes the suffix if there is one